packages/plugin-ma-tracking/src/ma-tracking.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  AdditionalBrowserOptions,
+  BaseEvent,
+  BeforePlugin,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import type { AmplitudeBrowser } from '@amplitude/analytics-browser';
+import { AdvancedCampaignTracker } from './advanced-campaign-tracker';
+
+export class MATrackingPlugin implements BeforePlugin {
+  name = 'MATracking';
+  type = PluginType.BEFORE as const;
+
+  constructor(private readonly instance: AmplitudeBrowser, private readonly options: AdditionalBrowserOptions = {}) {}
+
+  async setup(config: BrowserConfig): Promise<undefined> {
+    const attributionConfig = this.options.attribution ?? {};
+    const pageTrackingConfig =
+      this.options.trackPageViews ??
+      (attributionConfig.trackPageViews && {
+        filter: 'onAttribution',
+      });
+
+    const instance = this.instance;
+    // Share cookie storage with user session storage
+    const storage = config.cookieStorage as unknown as Storage<Campaign>;
+
+    const campaignTracker = new AdvancedCampaignTracker(config.apiKey, {
+      ...attributionConfig,
+      trackPageViews: false,
+      track: /* istanbul ignore next */ () => Promise.resolve(),
+      onNewCampaign: /* istanbul ignore next */ () => () => undefined,
```

Comment:
hmm looks like these are only added for backwards compatibility? maybe we can just create a separate campaign tracker and not extend the old one

Reply:
It is simpler to extend the existing one because both page tracking and web attribution plugins use the same functionality. 

Or make these two functions optional and add default values to these properties.